hphp/hack/test/hackfmt/tests/allow_single_line_ifs.php

```diff
@@ -0,0 +1,82 @@
+<?hh // strict
+
+// Should be on a single line
+if (true) print("hi");
+
+// Should not be on a single line
+if (true)
+print("hi");
+
+if (true)
+  print("hi");
+
+if (true) {
+print("this is a");
+print("multi line if");
+}
+
+if (true)
+print("this is not a");
+print("multi line if");
+
+// Should all have the if body on a new line
+if (true) this_is_a_very_long_function_call_that_really_exceeds_the_line_length_lorem_ipsum_dolor_sit_amet_consectetur_adipiscing_elit();
+
+if (true)
+this_is_a_very_long_function_call_that_really_exceeds_the_line_length_lorem_ipsum_dolor_sit_amet_consectetur_adipiscing_elit();
+
+if (true)
+  this_is_a_very_long_function_call_that_really_exceeds_the_line_length_lorem_ipsum_dolor_sit_amet_consectetur_adipiscing_elit();
+
+// Should preserve the single line if
+f(g($a, $b), () ==> {
+  if (true) print('hi');
```

Comment:
Can we also add a test here where we have a multiline `if` statement that's inside a block?  Something like this:

```
  if (true) print('hi');
  if (true)
  print('hi');
```

This way we can ensure that the parental rule is activating properly :)

Reply:
Thanks for the feedback! I added the multi-line if statement, as well as a single-line if statement inside a multi-line if statement below